### PR TITLE
Support asynchronous file I/O on Windows

### PR DESCRIPTION
**Pull request by HertzDevil**
Since #14255, it is now possible to open non-blocking `File`s on Windows, as previously their blocking mode could not be changed after initialization. This PR makes operations on those `File`s actually work.

### CODE REVIEW

File: src/crystal/system/win32/file_descriptor.cr [hunk 1]
```diff
@@ -200,19 +201,12 @@ module Crystal::System::FileDescriptor
...
-    overlapped.union.offset.offset = LibC::DWORD.new(offset)
-    overlapped.union.offset.offsetHigh = LibC::DWORD.new(offset >> 32)
-    if LibC.ReadFile(handle, buffer, buffer.size, out bytes_read, pointerof(overlapped)) == 0
-      error = WinError.value
-      return 0_i64 if error == WinError::ERROR_HANDLE_EOF
-      raise IO::Error.from_os_error "Error reading file", error, target: self
-    end
-
-    bytes_read.to_i64
+  def self.pread(file, buffer, offset)
```
Comment (yxhuvud) [inline src/crystal/system/win32/file_descriptor.cr:204]: It is high time to rename this method as the name refers to a libc call that isn't used on this OS, which is very much not asynchronous. 

`read_with_offset` perhaps?

**Review by yxhuvud — COMMENTED** [review]
Nice. Is this the first case of windows functionality being better than the libevent based one?

**Comment by beta-ziliani** [comment]
@HertzDevil is this ready for review?

**Comment by HertzDevil** [comment]
All the progress here has been split into individual PRs that work with the refactored IOCP interface; closing

**Pull request closed without merging**